src/tss2-esys/esys_crypto_ossl.c

```diff
@@ -597,6 +604,7 @@ iesys_cryptossl_pk_encrypt(TPM2B_PUBLIC * pub_tpm_key,
     if (1 != RSA_generate_key_ex(rsa_key,
                                  pub_tpm_key->publicArea.parameters.rsaDetail.keyBits,
                                  bne, NULL)) {
+        RAND_set_rand_method(rand_save);
```

Comment:
Don't need that here as this is done in the cleanup block, but we do need this in line 566 before the return.

Reply:
So true